### DISCUSSION TITLE
Remix support

### DISCUSSION THREAD

**Discussion by niklasgrewe** [markdoc/markdoc#86] [Ideas]
Hi, thanks a lot for this library. I use Remix for the development of my website. Currently there is only one plugin for Nextjs. Will there be plugins for other frameworks like Remix in the future?

How can I use markdoc in Remix today so that I can create `.md` files, any examples how can i do that?

**Comment by mfix-stripe** [comment]
@niklasgrewe Remix is a great idea! I think it will be pretty straightforward to create a Remix plugin that integrates Markdoc. If anyone is interested in doing so, you can probably follow the code from our Next.js plugin: https://github.com/markdoc/next.js.

**Comment by sergiodxa** [comment]
Since you can't customize Remix compiler, it shouldn't be possible (without patch-package) to make Remix use Markdoc to render `.md` files or `.mdoc` files that you use in your routes folders or you import in your other routes.

Also, the recommended way to work with Markdown in Remix is to treat the MD content as data and not code (even if you can import MD files), so you should use `fs.readFile` to read it from the file system or store the MD content in a DB and query it. In either case, you can use Markdoc `parse`, `transform` and `renderers.react` without any extra plugin.

[I wrote about this](https://sergiodxa.com/articles/parse-markdown-with-markdoc-in-remix) but here's a simple example usage inside a Route:

```tsx
import { json, type LoaderArgs } from "@remix-run/node";
import { useLoaderData } from "@remix-run/react";
import { parse, transform, renderers } from "@markdoc/markdoc";

export function loader({ params }: LoaderArgs) {
  // here get the MD from your fs using or a database
  let markdown = await getArticleContent(params.article);

  let ast = Markdoc.parse(doc);
  let content = content = Markdoc.transform(ast);
  return json({ content });
}

export default function Route() {
  let { content } = useLoaderData<typeof loader>();
  return <>{Markdoc.renderers.react(content, React)}</div>
}
```

**Comment by mfix-stripe** [comment]
@sergiodxa @niklasgrewe I do wonder if Remix would be open to someone adding native Markdoc support to Remix though (for docs ending in `.mdoc`). 

Here is the plugin for MDX: https://github.com/remix-run/remix/blob/main/packages/remix-dev/compiler/plugins/mdx.ts

**Reply by sergiodxa** [reply]
AFAIk they actually want to remove MDX support, the recommended way to work with Markdown in Remix is to treat the MD as data and not code, so your loader functions should get the MD content from somewhere (file system, API, database, cache, anything) and then parse it, which btw Markdoc makes super easy to do.

**Reply by mfix-stripe** [reply]
Oh fantastic! That's actually great to hear. Aligns well w/ Markdoc's patterns, too.
